### PR TITLE
fix: improve onebox work dirs

### LINKED ISSUE #1328: scripts: improve onebox script
https://github.com/4paradigm/OpenMLDB/blob/1aecdbc7f3bffe62373fecdab158f9f6611d9775/onebox/start_onebox.sh#L35-L36

The onebox start script will create all dirs and logs in the root dir. It's hard to cleanup.
Plz move it to another place, and each server should have an unique log path(use `-openmldb_log_dir=xxx`).

### PR DESCRIPTION
**Pull request by auula**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

1. fix [https://github.com/4paradigm/OpenMLDB/issues/1328](https://github.com/4paradigm/OpenMLDB/issues/1328)

* **What is the current behavior?** (You can also link to an open issue here)
2.  The log output by the onebox script is not in the unified directory.

* **What is the new behavior (if this is a feature change)?**
3.  The subdirectory can be set through the environment variable ONEBOX_WORKDIR.
4.  Log unified directory.

### CODE REVIEW

File: onebox/start_onebox.sh [hunk 1]
```diff
@@ -19,11 +19,22 @@ set -x -e
...
+# rm -rf logs/
+# mkdir -p logs
+
+# The subdirectory can be set through the environment variable ONEBOX_WORK
+workspace=$ONEBOX_WORK
 
 IP=127.0.0.1
 
+# work unified file directory
+if [ ! -n "$workspace" ]; then
```
Comment (aceforeverd) [inline onebox/start_onebox.sh]: pls resolve shellcheck errors

Comment (aceforeverd) [inline onebox/start_onebox.sh]: I recommend `workspace` default to a subdirectory inside `onebox/`

File: onebox/start_onebox.sh [hunk 2]
```diff
@@ -19,11 +19,22 @@ set -x -e
...
+  # default work directory
+  workspace="workspace"
+  rm -rf $workspace/logs/
+  mkdir -p $workspace/logs
```
Comment (aceforeverd) [inline onebox/start_onebox.sh]: should mkdir for other `workspace` too

**Review by aceforeverd — CHANGES_REQUESTED** [review]

File: onebox/start_onebox.sh (hunk 2 above)
Comment (auula) [reply onebox/start_onebox.sh:35]: ok

File: onebox/start_onebox.sh [hunk 3]
```diff
@@ -19,11 +19,22 @@ set -x -e
 cd "$(dirname "$0")/../"
 
 ulimit -c unlimited
-rm -rf logs/
-mkdir -p logs
+# rm -rf logs/
+# mkdir -p logs
```
Comment (vagetablechicken) [inline onebox/start_onebox.sh]: remove the comments, line 22-23

File: onebox/start_onebox.sh [hunk 4]
```diff
@@ -19,11 +19,22 @@ set -x -e
...
+
+# The subdirectory can be set through the environment variable ONEBOX_WORK
+workspace=$ONEBOX_WORK
```
Comment (vagetablechicken) [inline onebox/start_onebox.sh]: You can use `WORKSPACE=${ONEBOX_WORKDIR:-onebox/workspace}`
No need to do `if ! -n "$workspace"`

File: onebox/start_onebox.sh (hunk 1 above)
Comment (auula) [reply onebox/start_onebox.sh]: solved.

File: onebox/start_onebox.sh (hunk 4 above)
Comment (auula) [reply onebox/start_onebox.sh]: solved.

Comment (auula) [reply onebox/start_onebox.sh]: review code ：[https://github.com/4paradigm/OpenMLDB/pull/1899/commits/760ff33219d39661801bbcd212dd7ba8830a8954](https://github.com/4paradigm/OpenMLDB/pull/1899/commits/760ff33219d39661801bbcd212dd7ba8830a8954)

File: onebox/start_onebox.sh (hunk 1 above)
Comment (auula) [reply onebox/start_onebox.sh]: review code ：[https://github.com/4paradigm/OpenMLDB/pull/1899/commits/760ff33219d39661801bbcd212dd7ba8830a8954](https://github.com/4paradigm/OpenMLDB/pull/1899/commits/760ff33219d39661801bbcd212dd7ba8830a8954)

File: onebox/start_onebox.sh (hunk 2 above)
Comment (auula) [reply onebox/start_onebox.sh]: review code ：[https://github.com/4paradigm/OpenMLDB/pull/1899/commits/760ff33219d39661801bbcd212dd7ba8830a8954](https://github.com/4paradigm/OpenMLDB/pull/1899/commits/760ff33219d39661801bbcd212dd7ba8830a8954)

File: onebox/start_onebox.sh (hunk 3 above)
Comment (auula) [reply onebox/start_onebox.sh]: remove comments: [https://github.com/4paradigm/OpenMLDB/pull/1899/commits/c547f446a508b080b9d27517d66d8e72e0cbd9f3](https://github.com/4paradigm/OpenMLDB/pull/1899/commits/c547f446a508b080b9d27517d66d8e72e0cbd9f3)

Comment (auula) [reply onebox/start_onebox.sh]: > remove the comments, line 22-23

remove comments: [https://github.com/4paradigm/OpenMLDB/pull/1899/commits/c547f446a508b080b9d27517d66d8e72e0cbd9f3](https://github.com/4paradigm/OpenMLDB/pull/1899/commits/c547f446a508b080b9d27517d66d8e72e0cbd9f3)

File: onebox/start_onebox.sh (hunk 4 above)
Comment (auula) [reply onebox/start_onebox.sh]: > You can use `WORKSPACE=${ONEBOX_WORKDIR:-onebox/workspace}` No need to do `if ! -n "$workspace"`

review code ：[https://github.com/4paradigm/OpenMLDB/pull/1899/commits/760ff33219d39661801bbcd212dd7ba8830a8954](https://github.com/4paradigm/OpenMLDB/pull/1899/commits/760ff33219d39661801bbcd212dd7ba8830a8954)

File: onebox/start_onebox.sh (hunk 2 above)
Comment (auula) [reply onebox/start_onebox.sh]: > should mkdir for other `workspace` too

review code ：[https://github.com/4paradigm/OpenMLDB/pull/1899/commits/760ff33219d39661801bbcd212dd7ba8830a8954](https://github.com/4paradigm/OpenMLDB/pull/1899/commits/760ff33219d39661801bbcd212dd7ba8830a8954)

File: onebox/start_onebox.sh (hunk 1 above)
Comment (auula) [reply onebox/start_onebox.sh]: > I recommend `workspace` default to a subdirectory inside `onebox/`
review code ：[https://github.com/4paradigm/OpenMLDB/pull/1899/commits/760ff33219d39661801bbcd212dd7ba8830a8954](https://github.com/4paradigm/OpenMLDB/pull/1899/commits/760ff33219d39661801bbcd212dd7ba8830a8954)

**Review by vagetablechicken — APPROVED** [review]

**Review by aceforeverd — APPROVED** [review]

**Pull request merged**